pkgs/by-name/ig/igir/package.nix

```diff
@@ -2,46 +2,87 @@
   lib,
   buildNpmPackage,
   fetchFromGitHub,
-
-  # for patching bundled 7z binary from the 7zip-bin node module
-  # at lib/node_modules/igir/node_modules/7zip-bin/linux/x64/7za
-  autoPatchelfHook,
   stdenv,
+  autoPatchelfHook,
+  SDL2,
+  libuv,
+  lz4,
+  #tests
+  runCommand,
+  igir,
+  nodejs,
 }:
-
 buildNpmPackage rec {
   pname = "igir";
-  version = "2.11.0";
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     owner = "emmercm";
     repo = "igir";
     rev = "v${version}";
-    hash = "sha256-NG0ZP8LOm7fZVecErTuLOfbp1yvXwHnwPkWTBzUJXWE=";
+    hash = "sha256-QPUnvirufY3UiSFVxX3xCmpAlzPKg5JxwAnVefdepqU=";
   };
 
-  npmDepsHash = "sha256-ADIEzr6PkGaJz27GKSVyTsrbz5zbud7BUb+OXPtP1Vo=";
+  npmDepsHash = "sha256-miXxMBtIqgxS4Ulhkq79VDCkBjDjoB8juPoDWE1UQUM=";
+
+  # This patch is only needed for 3.0.1 and needs to be removed afterwards
+  # along with the patch file itself
+  patches = [ ./0001-update-dependencies.patch ];
 
   # I have no clue why I have to do this
   postPatch = ''
     patchShebangs scripts/update-readme-help.sh
   '';
 
-  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [
+    (lib.getLib stdenv.cc.cc)
+    SDL2
+    libuv
+    lz4
```

Comment:
Weren't these two only required on Linux?

Reply:
@eljamm That's what I incorrectly said: while the autopatchelf hook finds all binaries, I had to manually patch them for darwin. And I initially missed maxcso, which is the component requiring those dependencies.